python/pyspark/rdd.py

```diff
@@ -751,7 +751,7 @@ def func(iterator):
 
             def pipe_objs(out):
                 for obj in iterator:
-                    s = str(obj).rstrip('\n') + '\n'
+                    s = unicode(obj).rstrip('\n') + '\n'
```

Comment:
@chaoslawful, if you are active, we could change `\n` to `u\n` to reduce the conversion and don't rely on the implicit conversion between `str` and `unicode`.